### PR TITLE
Set the thread name to the currently-executing task's name.

### PR DESCRIPTION
**Pull request by swankjesse**
Also implement tests for thread interruption.
Also add a test that uses the real backend.

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/internal/concurrent/Task.kt [hunk 1]
```diff
@@ -67,20 +67,30 @@ abstract class Task(
...
       try {
         delayNanos = runOnce()
       } finally {
         queue.runCompleted(this, delayNanos)
+        currentThread.name = oldName
       }
     }
 
     this.cancelRunnable = Runnable {
+      val currentThread = Thread.currentThread()
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/concurrent/Task.kt]: Why not use the inline function Util.threadName?

File: okhttp/src/main/java/okhttp3/internal/concurrent/TaskRunner.kt [hunk 2]
```diff
@@ -161,5 +160,10 @@ class TaskRunner(
     override fun coordinatorWait(taskRunner: TaskRunner, nanos: Long) {
       taskRunner.objectWaitNanos(nanos)
     }
+
+    fun shutDown() {
+      coordinatorExecutor.shutdown()
+      coordinatorExecutor.shutdown()
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/concurrent/TaskRunner.kt]:
```suggestion
      taskExecutor.shutdown()
```

File: okhttp/src/main/java/okhttp3/internal/concurrent/Task.kt (hunk 1 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/concurrent/Task.kt]: I’m hoping to delete it!

File: okhttp/src/main/java/okhttp3/internal/concurrent/TaskRunner.kt (hunk 2 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/concurrent/TaskRunner.kt]: Whoa embarrassing mistake!

**Pull request merged**